sql/core/src/test/scala/org/apache/spark/sql/TimestampTypeSuite.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import java.sql.Timestamp
+import java.time.LocalDateTime
+
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.{StructField, StructType, TimestampNTZType, TimestampType}
+import org.apache.spark.util.ResetSystemProperties
+
+// Test suite for setting the default timestamp type of Spark SQL
+class TimestampTypeSuite extends QueryTest with SharedSparkSession with ResetSystemProperties {
```

Comment:
We already have `DataTypeSuite`, so, could put the tests there, or you plan so many tests for TimestampType?

Reply:
I think we can simply write tests in `DataTypeParserSuite`

Reply:
if we apply the config in timestamp literals later, we should add tests in `ExpressionParserSuite`

Reply:
This is end-to-end test with actual queries, which can't be done in DataTypeSuite.
I just renamed the file to `SetDefaultTimestampTypeSuite` to avoid the confusion.

Reply:
@cloud-fan oh I missed your comments. Let me write tests in DataTypeParserSuite